src/cmd/flux-job.c

```diff
@@ -829,6 +844,39 @@ void attach_output_continuation (flux_future_t *f, void *arg)
         }
         free (data);
     }
+    else if (!strcmp (name, "redirect")) {
+        const char *stream = NULL;
+        const char *rank = NULL;
+        const char *path = NULL;
+        struct idset *idset = NULL;
+        if (!ctx->output_header_parsed)
+            log_msg_exit ("stream redirect read before header");
+        if (json_unpack (context, "{ s:s s:s s?:s }",
+                         "stream", &stream,
+                         "rank", &rank,
+                         "path", &path) < 0)
+            log_msg_exit ("malformed redirect context");
+        if (!(idset = idset_decode (rank)))
+            log_msg_exit ("malformed rank idset '%s'", rank);
+        if (!strcmp (stream, "stdout"))
+            ctx->stdout_redirect_count += idset_count (idset);
+        else
+            ctx->stderr_redirect_count += idset_count (idset);
```

Comment:
It seems like this code shouldn't actually need to count redirects and cancel the RPC.  The RPC stream will eventually end with ENODATA when the job ends.

Since we're ignoring EOFs it seems like we can ignore these too, except for printing the warning message.

The block comment at the top of the function should be updated to mention redirect events.

Reply:
ahh yes, i had done things a little differently in a earlier incarnation of the code.